xbmc/windowing/X11/WinSystemX11.cpp

```diff
@@ -1070,4 +1072,49 @@ void CWinSystemX11::UpdateCrtc()
   g_graphicsContext.SetFPS(fps);
 }
 
+std::string CWinSystemX11::GetClipboardText(void)
+{
+  if (!m_dpy)
+     return "";
+
+  Atom utf8_string = XInternAtom(m_dpy, "UTF8_STRING", False);
+  if (utf8_string == None)
+    return "";
+
+  Atom clipboard = XInternAtom(m_dpy, "CLIPBOARD", False);
+  if (clipboard == None)
+    return "";
+
+  // if no selection or there is no owner for the selection (ancient X ?!)
+  if (XGetSelectionOwner(m_dpy, clipboard) == None)
+    return "";
+
+  std::string result = "";
+  XConvertSelection(m_dpy, clipboard, utf8_string, None, m_glWindow, CurrentTime);
+  CWinEventsX11Imp::MessagePump();
+
+  for(unsigned long offset = 0;;)
```

Comment:
no chance for this loop, as I said: wait for the specific event for a maximum time.
and again: follow the example of SDL